cpp/dolfinx/fem/utils.cpp

```diff
@@ -170,59 +184,104 @@ fem::compute_integration_domains(fem::IntegralType integral_type,
       throw std::runtime_error(
           "Topology cell-to-facet connectivity has not been computed.");
     }
+    return {f_to_c, c_to_f};
+  };
 
-    switch (integral_type)
+  auto get_cell_vertex_connectivity = [tdim, &topology]()
+      -> std::pair<std::shared_ptr<const graph::AdjacencyList<int>>,
+                   std::shared_ptr<const graph::AdjacencyList<int>>>
+  {
+    auto v_to_c = topology.connectivity(0, tdim);
+    if (!v_to_c)
     {
-    case IntegralType::exterior_facet:
+      throw std::runtime_error(
+          "Topology vertex-to-cell connectivity has not been computed.");
+    }
+
+    auto c_to_v = topology.connectivity(tdim, 0);
+    if (!c_to_v)
     {
-      // Create list of tagged boundary facets
-      const std::vector bfacets = mesh::exterior_facet_indices(topology);
-      std::vector<std::int32_t> facets;
-      std::ranges::set_intersection(entities, bfacets,
-                                    std::back_inserter(facets));
-      for (auto f : facets)
-      {
-        // Get the facet as a pair of (cell, local facet)
-        auto facet
-            = impl::get_cell_facet_pairs<1>(f, f_to_c->links(f), *c_to_f);
-        entity_data.insert(entity_data.end(), facet.begin(), facet.end());
-      }
+      throw std::runtime_error(
```

Comment:
I do not think this should ever happen, as it is required to initialize the Topology (post the last refactoring):
https://github.com/FEniCS/dolfinx/blob/main/cpp/dolfinx/mesh/Topology.h#L56-L57
I don't think we at the moment are able to delete connectivity (even if it was made so by design).
So I guess this doesn't hurt.